### DISCUSSION TITLE
Device Management Port / IP

### DISCUSSION THREAD

**Discussion by JHarrison712** [netbox-community/netbox#12042] [Ideas]
Good morning everyone,

We usually custom each of our switch ports or firewall or devices so its not standard. I noticed here on my device page, that I think it would be useful to have a Management Port line, maybe replace Primary IPv6. Also might allow Secondary IPv4 in the same list say for a firewall that has a second connection for management or etc.

**Comment by candlerb** [comment]
By "management port" do you mean TCP/UDP port? In that case do you mean SSH, HTTPS, SNMP, others... ? For these, have a look at the "Services" model (go to the device detail view, look at the Services section, click Add Service). You can use "Custom" instead of "From Template" if the port number is different for every device.  Another option is to use Custom Fields.

If you mean a management physical port with its own management (out-of-band) IP address, you can set the "Management Only" flag on an interface. You can then pick up the management IP address(es) by scanning for management interface(s) and the address(es) assigned to those - admittedly not as easy as seeing the "Primary IPv4/IPv6" addresses directly.

If the rule is that a device can only have one out-of-band management address, then you can make a Custom Link like this:

```
### Link Text
{% if object.interfaces.filter(mgmt_only=True).count() == 1 and object.interfaces.get(mgmt_only=True).ip_addresses.count() == 1 %}Open IPMI{% endif %}

### Link URL
https://{{ object.interfaces.get(mgmt_only=True).ip_addresses.first().address.ip }}/
```
